relay-server/src/actors/project_upstream.rs

```diff
@@ -421,15 +440,27 @@ impl UpstreamProjectSourceService {
         });
     }
 
-    /// Handles the incoming external messages.
     fn handle_message(&mut self, message: UpstreamProjectSource) {
-        let UpstreamProjectSource(
-            FetchProjectState {
-                project_key,
-                no_cache,
-            },
-            sender,
-        ) = message;
+        match message {
+            UpstreamProjectSource::FetchProjectState(fetch_project_state, sender) => {
+                self.handle_project_source(fetch_project_state, sender)
+            }
+            UpstreamProjectSource::FetchGlobalConfig(global_config) => {
+                self.global_config = global_config;
```

Comment:
I think we need to reconsider allowing different global configs to exist at the same time. Project states may live up to one hour and that's a long time for the global config to go out of date, which will lead to behavior that's hard to explain.

I think we should make this an `Arc<Mutex>` or better yet, publish it not into the project cache but into the service that actually needs it (the `EnvelopeProcessor`). We can still have a somewhat unified interface that is created inside the `EnvelopeProcessor` on the fly.